### ISSUE TITLE
Suggestion: Allowing custom columns in nearest_nodes function

### ISSUE THREAD

**Issue by gianmarconaro** [gboeing/osmnx#975]
**Read these instructions carefully. Use the template below and fill out every section. Your issue will be automatically closed if you don't provide all of the information requested in the template, because we need this information to address your issue. Before you proceed, review the contributing guidelines in the CONTRIBUTING.md file.**

Feature requests are for proposing an enhancement to OSMnx before opening a pull request to add your proposed functionality to the codebase. Please don't use the issue tracker to ask "how-to" or usage questions. If asked here, they will be closed automatically. Instead, ask "how-to" and usage questions on [StackOverflow](https://stackoverflow.com/search?q=osmnx).

**Completely fill out the following template.**

  - [x] I have read the OSMnx [contributing guidelines](../blob/main/CONTRIBUTING.md)
  - [x] I have read the OSMnx [documentation](https://osmnx.readthedocs.io/) and [usage examples](https://github.com/gboeing/osmnx-examples)
  - [x] I have searched the open/closed issues and pull requests for others similar to mine

**Is your feature proposal related to a problem?**

Hello,

I am finding it very useful to use your library even on custom networkx graphs, and I have a suggestion regarding the `nearest_nodes` function. The current implementation of the `nearest_nodes` function hardcodes the column names as "x" and "y" for latitude and longitude, respectively. However, it would be beneficial to allow users to specify their own column names for latitude and longitude, providing more flexibility and customization options for users. 

The `nearest_nodes` function currently assumes that the latitude and longitude coordinates are stored in the "x" and "y" columns, respectively.

```python
nodes = utils_graph.graph_to_gdfs(G, edges=False, node_geometry=False)[["x", "y"]]
```

**Describe the solution you'd like to propose**

I think that is possibile to modify the `nearest_nodes` function to allow users to specify custom column names as `latitude` and `longitude`. This enhancement would enable users to use other column names that suit their dataset structure and the function would become more versatile and adaptable to different datasets.

A possible implementation idea could be to use a dictionary for attributes (similar to how networkx adds nodes or edges to a graph).

**Benefits:**

-   Enhanced flexibility and customization options for users.
-   Improved compatibility with datasets that have different column naming conventions.
-   Reduced need for manual preprocessing or renaming of columns.

**Describe alternatives you've considered**

The alternative I have considered is to align myself and modify the columns of my graph so that I can use your functions.

**Additional context**

I retrieve my information regarding latitude and longitude with a query and then I create the nodes of the graph in this way:

``` python
G.add_node(id, lon=query[2], lat=query[1], attr=...)
```
If I generate my graph in this way, the function will return a `key error` due to the column names. Changing the column names in "x" and "y", of course, resolves the issue.

Thank you for your attention and support!

**Comment by gboeing** [comment]
Thanks for using OSMnx! I have mixed feelings with this kind of suggestion. On one hand, I agree that OSMnx ideally should be as useful to as many people as possible. On the other hand, this suggestion would be outside the scope and mission of OSMnx (modeling and analyzing OSM APIs' data in a consistent way). It also would introduce inconsistencies where the standard expected OSMnx `x`/`y` attributes are customizable in the `nearest_nodes` function but not elsewhere throughout the package where they are expected.

If this created an obstacle that *prevented* a set of important use cases, I'd be inclined to consider making a broader change. But going beyond the scope and mission of OSMnx creates a more cluttered API, more technical debt, and more ongoing codebase maintenance work (for me). Accordingly, the philosophy of OSMnx has been to support flexibility in that users can adjust their data structures (GeoDataFrames or MultiDiGraphs) to work with OSMnx's functionality.

For example, in this case, the user can rename node attributes like:

```python
def rename_attrs(G, mapper):
    for _, d in G.nodes(data=True):
        for k, v in mapper.items():
            d[v] = d.pop(k)
    return G
rename_attrs(G, {"lng": "x", "lat": "y"})  # lng/lat -> x/y
rename_attrs(G, {"x": "lng", "y": "lat"})  # and back again
```

So, long story short, I agree with you that it would make things easier for this use case if you could specify different attributes as `x`/`y`, but given that there's a straightforward workaround, I'm inclined to maintain the current functionality.

**Comment by gianmarconaro** [comment]
Thank you for your response. I understand your perspective, and in this scenario, it makes sense not to make significant changes since the issue is not overly obstructive and it would go against philosophy of OSMnx. I simply wanted to bring it up because I'm using OSMnx for my master's thesis and have found it immensely valuable, especially for custom graphs that are linked with OSM, such as those derived from GTFS data. Consequently, I thought it would be beneficial to propose a way to enhance the library's flexibility. Moreover, I truly appreciate the suggestion you provided, and I believe it will be useful. Thank you for dedicating your time and your support!
